### PR TITLE
Initial enviroment setup using webpack

### PR DESCRIPTION
**Pull request by nomad302**
Initial enviroment setup using webpack for Demo React typescript webpack app.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+<head>
+    <meta charset="UTF-8" />
+    <title>Hello React Typescript!</title>
+</head>
+
+<body>
+    <div id="demo-app"></div>
+
+    <!-- Dependencies -->
```
Comment (hrc7505) [inline index.html:12]: Is this really needed? I don't think this is needed because without this it should work.

File: package.json [hunk 2]
```diff
@@ -4,14 +4,31 @@
...
     "type": "git",
     "url": "git+https://github.com/nomad302/demo-app-ts-react.git"
   },
   "author": "\"sagar\"",
   "license": "ISC",
+  "devDependencies": {
+    "@types/react": "^16.9.36",
+    "@types/react-dom": "^16.9.8",
+    "awesome-typescript-loader": "^5.2.1",
+    "contentful": "^7.14.4",
```
Comment (hrc7505) [inline package.json:19]: You can move this to dependencies. It is not a development dependency.

File: src/components/App.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import * as React from 'react';
+
+export interface HelloProps { greet: string; company: string }
```
Comment (hrc7505) [inline src/components/App.tsx:3]: Semicolon should be there. We are following these rules so. linting will give you an error for this but it is still not configured so I am adding a comment.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    <script src="./node_modules/react/umd/react.development.js"></script>
+    <script src="./node_modules/react-dom/umd/react-dom.development.js"></script>
+
+    <!-- Main -->
+    <script src="./dist/demo-app.js"></script>
+</body>
+
+</html>
```
Comment (hrc7505) [inline index.html:20]: New line should be at the end of every file. This is one of the linting rule and we are following it.

File: src/index.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import { App } from './components/App'
+
+
```
Comment (hrc7505) [inline src/index.tsx:6]: Unwanted extra new line should be removed. Only one new line is fine.

File: src/index.tsx [hunk 6]
```diff
@@ -0,0 +1,7 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+import { App } from './components/App'
```
Comment (hrc7505) [inline src/index.tsx:4]: Configuration should be for path like `components/App `. We always start import path from the folders directly located in `src`.

File: index.html (hunk 1 above)
Comment (nomad302) [reply index.html:12]: Dependencies??

File: package.json (hunk 2 above)
Comment (nomad302) [reply package.json:19]: only this dependency "contentful": "^7.14.4", right?

File: index.html (hunk 1 above)
Comment (hrc7505) [reply index.html:12]: Yes. both script links.

File: package.json (hunk 2 above)
Comment (hrc7505) [reply package.json:19]: Yes

File: src/components/App.tsx (hunk 3 above)
Comment (nomad302) [reply src/components/App.tsx:3]: Yes !! I will be configuring it in another branch!!

File: index.html (hunk 1 above)
Comment (nomad302) [reply index.html:12]: Ok! Let me check !!

File: package.json (hunk 2 above)
Comment (nomad302) [reply package.json:19]: Thanks!!

File: index.html (hunk 4 above)
Comment (nomad302) [reply index.html:20]: Okay !! Will do it !!

File: src/components/App.tsx [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import * as React from 'react';
+
+export interface HelloProps { greet: string; company: string };
```
Comment (hrc7505) [inline src/components/App.tsx:3]: Make a separate folder for interfaces. We will follow structure like,
```
- src
   - index.tsx
   - app.tsx
   - interfaces (contains interfaces for props and state or any other related only to app.tsx)
      - IAppProps.ts
      - IAppState.ts
   - home-page
      - interfaces
      - HomePageComponent
      - homePageStyle.css
```
So create a new folder for interfaces and move this interface to that folder.

Comment (nomad302) [reply src/components/App.tsx:3]: Okay, Will do, Thanks!

File: src/App.tsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import * as React from 'react';
+import IHelloProps from 'interfaces/IAppProps'
+
+export default class App extends React.Component<IHelloProps, {}>{
```
Comment (hrc7505) [inline src/App.tsx:4]: Why you changed this to class component?

File: src/interfaces/IAppProps.ts [hunk 9]
```diff
@@ -0,0 +1 @@
+export default interface HelloProps { greet: string; company: string; }
```
Comment (hrc7505) [inline src/interfaces/IAppProps.ts:1]: formatting is required! Readability matters! Please fix this.
```
export default interface HelloProps { 
        greet: string; 
        company: string; 
}
```

File: src/App.tsx (hunk 8 above)
Comment (nomad302) [reply src/App.tsx:4]: Actually, I just thought that in future we need an App component with the state, so change it here.

File: src/interfaces/IAppProps.ts (hunk 9 above)
Comment (nomad302) [reply src/interfaces/IAppProps.ts:1]: Okay, I Will Do it, Thanks.

File: src/App.tsx [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+    constructor(props: IHelloProps) {
```
Comment (hrc7505) [inline src/App.tsx:5]: I don't think `constructor` is needed here.

Comment (nomad302) [reply src/App.tsx:5]: Yes, I will resolve it. Thanks!

File: src/App.tsx (hunk 8 above)
Comment (hrc7505) [reply src/App.tsx:4]: If needed in future we can change it later.

Comment (nomad302) [reply src/App.tsx:4]: Okay, thanks.

**Pull request merged**